ground-control/internal/server/handlers.go

```diff
@@ -369,39 +410,15 @@ func (s *Server) registerSatelliteHandler(w http.ResponseWriter, r *http.Request
 			tx.Rollback()
 			return
 		}
-
-	}
-
-	// Create the satellite's state artifact
-	err = utils.CreateOrUpdateSatStateArtifact(r.Context(), req.Name, groupStates)
-	if err != nil {
-		log.Println(err)
-		tx.Rollback()
-		HandleAppError(w, err)
-		return
 	}
 
-	// Add token to DB
-	token, err := GenerateRandomToken(32)
-	if err != nil {
-		log.Println(err)
-		tx.Rollback()
-		HandleAppError(w, err)
-		return
-	}
-	tk, err := q.AddToken(r.Context(), database.AddTokenParams{
-		SatelliteID: satellite.ID,
-		Token:       token,
-	})
-	if err != nil {
-		log.Println("error in token")
-		log.Println(err)
+	transactionSuccess = true
+	if err = tx.Commit(); err != nil {
+		log.Println("Commit failed:", err)
```

Comment:
While testing this when fail to commit to db. robot account is left intact. remove the robot account even on failure to commit
```suggestion
	transactionSuccess = true
	if err = tx.Commit(); err != nil {
		log.Println("Commit failed:", err)
		transactionSuccess = true
```